### PR TITLE
[CFM-1208] Theming for document file upload

### PR DESCRIPTION
**Pull request by guytepper**
Changed the file field of the document content type to be an entity reference to `File` and the widget to `C4M Document`.  

**Before:**

**After:**
after

**JIRA:** https://issuetracker.amplexor.com/jira/browse/CFM-1208

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_document/c4m_content_document.features.field_base.inc [hunk 1]
```diff
@@ -95,5 +72,40 @@ function c4m_content_document_field_default_field_bases() {
     'type' => 'entityreference',
   );
 
+  // Exported field_base: 'field_c4m_document'.
+  $field_bases['field_c4m_document'] = array(
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_document/c4m_content_document.features.field_base.inc]: @guytepper 
This one seems wrong, should be `c4m_document` without the `field_` prefix

File: project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php [hunk 2]
```diff
@@ -6,22 +6,24 @@
  */
 ?>
 <div class="form-group input-wrapper file-wrapper" id="add-document-form">
-  <label class="control-label"><?php print $title; ?></label>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]: @guytepper 
Did you make any changes over here ??
it doesn't seems like ...

Comment (guytepper) [inline project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]: Yup, I've [inlined](https://github.com/capacity4dev/capacity4more/pull/1119/commits/07cffc4a1023b7260d6c689c8949b7054f7f4348) required's `*` (asterisk) and [corrected](https://github.com/capacity4dev/capacity4more/pull/1119/commits/5198ee306c02445a89b255cbc52c4baa26aac27d) the indention.

**Comment by ordavidil** [comment]
@guytepper 
please re-roll

**Comment by guytepper** [comment]
@ordavidil locally on my machine the tests are passing. Any idea why travis fails running them?

**Comment by ordavidil** [comment]
There is a problem with the documents migration `C4mDemoContentCreateDocuments`

File: project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc [hunk 3]
```diff
@@ -102,6 +102,17 @@ class C4mDemoContentCreateDocuments extends C4mDemoContentCreateOgContent {
   public function prepareRow($row) {
     parent::prepareRow($row);
 
+    
+    // Saves the file to the database.
+    $new_file = $this->fileUriToObject(drupal_get_path('module', 'c4m_demo_content') . '/files/' . $row->file);
+
+    // Gets ID of the the newly stored file.
+    $fid = $new_file->fid;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc]: remove this line

File: project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc [hunk 4]
```diff
@@ -102,6 +102,17 @@ class C4mDemoContentCreateDocuments extends C4mDemoContentCreateOgContent {
   public function prepareRow($row) {
     parent::prepareRow($row);
 
+    
+    // Saves the file to the database.
+    $new_file = $this->fileUriToObject(drupal_get_path('module', 'c4m_demo_content') . '/files/' . $row->file);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc]: `$new_file` => `$file`

File: project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc [hunk 5]
```diff
@@ -102,6 +102,17 @@ class C4mDemoContentCreateDocuments extends C4mDemoContentCreateOgContent {
...
+
+    // Gets ID of the the newly stored file.
+    $fid = $new_file->fid;
+
+    // Change the migration file name to the file ID - since the migration is
+    // expecting an entity reference.
+    $row->file = $fid;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateDocuments.inc]:
```php
$row->file = $file->fid;
```

File: project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateEntity.inc [hunk 6]
```diff
@@ -485,4 +485,25 @@ abstract class C4mDemoContentCreateEntity extends Migration {
     return $unique;
   }
 
+  /**
+   * @param $uri
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/includes/CreateEntity.inc]: fix the docs

**Comment by ordavidil** [comment]
Closing in favor of #1297

**Pull request closed without merging**